owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/rendering/Renderer.java

```diff
@@ -652,5 +654,116 @@ public String render(final T value, Locale locale) {
+		return new RangeRenderer<C>(
+				rangeResourceKey, 
+				withoutUnitValueRenderer, 
+				(withUnitValueRenderer == null) ? withoutUnitValueRenderer : withUnitValueRenderer
+				).nullsAsNull();
+	}
+	
+	private static class RangeRenderer<C extends Comparable<?>> extends Renderer<Range<C>> {
+		private static final long serialVersionUID = -3069600849637042624L;
+		private final String resourceKeyPrefix;
+		private final Renderer<C> withUnitValueRenderer;
+		private final Renderer<C> withoutUnitValueRenderer;
+		
+		public RangeRenderer(String resourceKeyPrefix, Renderer<C> withoutUnitValueRenderer, Renderer<C> withUnitValueRenderer) {
+			this.resourceKeyPrefix = resourceKeyPrefix;
+			this.withoutUnitValueRenderer = withoutUnitValueRenderer;
+			this.withUnitValueRenderer = withUnitValueRenderer;
+		}
+		
+		@Override
+		public String render(final Range<C> value, Locale locale) {
+			IModel<Map<String, Object>> mapModel;
+			StringBuilder resourceKeyBuilder = new StringBuilder(resourceKeyPrefix);
+			if (!value.hasLowerBound() && !value.hasUpperBound()) {
+				return null;
+			} else if (!value.hasLowerBound()) {
+				resourceKeyBuilder.append(".upto");
+				mapModel = Models.dataMap()
+						.put("end", this.withoutUnitValueRenderer.asModel(Models.transientModel(value.upperEndpoint())))
+						.put("endUnit", this.withUnitValueRenderer.asModel(Models.transientModel(value.upperEndpoint())))
+						.build();
+			} else if (!value.hasUpperBound()) {
+				resourceKeyBuilder.append(".from");
+				mapModel = Models.dataMap()
+						.put("start", this.withoutUnitValueRenderer.asModel(Models.transientModel(value.lowerEndpoint())))
+						.put("startUnit", this.withUnitValueRenderer.asModel(Models.transientModel(value.lowerEndpoint())))
+						.build();
+			} else if (value.lowerEndpoint().equals(value.upperEndpoint())) {
+				resourceKeyBuilder.append(".solo");
+				mapModel = Models.dataMap()
+						.put("start", this.withoutUnitValueRenderer.asModel(Models.transientModel(value.lowerEndpoint())))
+						.put("startUnit", this.withUnitValueRenderer.asModel(Models.transientModel(value.lowerEndpoint())))
+						.put("end", this.withoutUnitValueRenderer.asModel(Models.transientModel(value.upperEndpoint())))
+						.put("endUnit", this.withUnitValueRenderer.asModel(Models.transientModel(value.upperEndpoint())))
+						.build();
+			} else {
+				resourceKeyBuilder.append(".both");
+				mapModel = Models.dataMap()
+						.put("start", this.withoutUnitValueRenderer.asModel(Models.transientModel(value.lowerEndpoint())))
+						.put("startUnit", this.withUnitValueRenderer.asModel(Models.transientModel(value.lowerEndpoint())))
+						.put("end", this.withoutUnitValueRenderer.asModel(Models.transientModel(value.upperEndpoint())))
+						.put("endUnit", this.withUnitValueRenderer.asModel(Models.transientModel(value.upperEndpoint())))
+						.build();
+			}
+			return Localizer.get().getString(resourceKeyBuilder.toString(), null, mapModel, locale, null, null);
```

Comment:
You may simply use the static method `Renderer.getString(String, Locale, IModel)` here.